lms/djangoapps/courseware/tests/test_date_summary.py

```diff
@@ -274,18 +278,71 @@ def test_course_end_date_after_course(self):
         )
         self.assertEqual(block.title, 'Course End')
 
-    ## Tests Verified Upgrade Deadline Date Block
+    # Tests Verified Upgrade Deadline Date Block
+
+    def check_upgrade_banner(self, banner_expected=True, include_url_parameter=True):
+        """
+        Helper method to check for the presence of the Upgrade Banner
+        """
+        url = reverse('info', args=[self.course.id.to_deprecated_string()])
+        if include_url_parameter:
+            url += '?upgrade=true'
+        resp = self.client.get(url)
+        if banner_expected:
+            self.assertIn("Give yourself an additional incentive to complete", resp.content)
+        else:
+            self.assertNotIn("Give yourself an additional incentive to complete", resp.content)
+
     @freeze_time('2015-01-02')
     def test_verified_upgrade_deadline_date(self):
-        self.setup_course_and_user(days_till_upgrade_deadline=1)
+        self.setup_course_and_user(days_till_upgrade_deadline=1, user_enrollment_mode=CourseMode.AUDIT)
         block = VerifiedUpgradeDeadlineDate(self.course, self.user)
         self.assertEqual(block.date, datetime.now(utc) + timedelta(days=1))
+        self.assertTrue(block.is_enabled)
         self.assertEqual(block.link, reverse('verify_student_upgrade_and_verify', args=(self.course.id,)))
 
     def test_without_upgrade_deadline(self):
         self.setup_course_and_user(enrollment_mode=None)
+        self.client.login(username='mrrobot', password='test')
         block = VerifiedUpgradeDeadlineDate(self.course, self.user)
+        self.assertFalse(block.is_enabled)
         self.assertIsNone(block.date)
+        self.check_upgrade_banner(banner_expected=False)
+
+    @freeze_time('2015-01-02')
+    def test_verified_upgrade_banner_present(self):
```

Comment:
To clarify, the banner only appears when logged in, but the RHS text appears even when not logged in (which is why this test is different from test_verified_upgrade_deadline_date)?